lib/Doctrine/ORM/Tools/EntityGenerator.php

```diff
@@ -1380,15 +1432,26 @@ protected function generateEntityStubMethod(ClassMetadataInfo $metadata, $type,
             $methodTypeHint =  '\\' . $typeHint . ' ';
         }
 
+        $methodReturnType = null;
+        if ($this->strictTypes) {
```

Comment:
+1 for two options, one for strict types and one for return types/scalar type hint